pkg/recipes/configloader/secrets.go

```diff
@@ -56,32 +60,49 @@ func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDResourceKe
 		}
 
 		// Populate the secretData map.
-		secretData, err = populateSecretData(secretStoreID, secretKeys, &secrets)
+		secretStoreData, err := populateSecretData(secretStoreID, secretKeysFilter, &secrets)
 		if err != nil {
 			return nil, err
 		}
+
+		// Merge the secret data for current secret store id into returned secretData map.
+		for key, value := range secretStoreData {
+			secretData[key] = value
+		}
 	}
 
 	return secretData, nil
 }
 
 // populateSecretData is a helper function to populate secret data from a secret store.
-func populateSecretData(secretStoreID string, secretKeys []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
-	secretData := make(map[string]map[string]string)
-
+// It takes a secret store ID, a filter for secret keys, and a response containing the secrets.
+// If the secret keys filter is nil or empty, it retrieves data for all keys in the secret store.
+// It returns a map where the keys are the secret store IDs and the values are maps of secret keys and their corresponding values.
+func populateSecretData(secretStoreID string, secretKeysFilter []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
 	if secrets == nil {
 		return nil, fmt.Errorf("secrets not found for secret store ID '%s'", secretStoreID)
 	}
 
-	for _, secretKey := range secretKeys {
-		if secretDataValue, ok := secrets.Data[secretKey]; ok {
-			if secretData[secretStoreID] == nil {
-				secretData[secretStoreID] = make(map[string]string)
-			}
-			secretData[secretStoreID][secretKey] = *secretDataValue.Value
-		} else {
+	secretData := make(map[string]map[string]string)
+
+	// If secretKeys is nil or empty, retrieve data for all keys
+	if len(secretKeysFilter) == 0 {
+		secretKeysFilter = make([]string, 0, len(secrets.Data))
+		for secretKey := range secrets.Data {
+			secretKeysFilter = append(secretKeysFilter, secretKey)
+		}
+	}
+
+	for _, secretKey := range secretKeysFilter {
+		secretDataValue, ok := secrets.Data[secretKey]
+		if !ok {
 			return nil, fmt.Errorf("a secret key was not found in secret store '%s'", secretStoreID)
 		}
+
+		if secretData[secretStoreID] == nil {
+			secretData[secretStoreID] = make(map[string]string)
+		}
```

Comment:
This one will be empty for sure, right? Why don't we move this to line 87 right after we initialize the slice? That would be more readable maybe?

Reply:
the return type is a map of maps. we do not want to return a map of keys (secretstoreids)with no data if no secret data exist. Keeping this as is for now.